src/test/xmeta/test_symbol_table.cpp

```diff
@@ -110,3 +110,176 @@ TEST_CASE("Enum circular dependency")
+        namespace N
+        {
+            enum E { }
+            enum F{}
+            delegate Int32 D1(Int32 i, Double d, E e);
+            delegate void D2();
+        }
+    )" };
+
+    xmeta_idl_reader reader{ "" };
+    REQUIRE(reader.read(test_idl) == 0);
+
+    auto namespaces = reader.get_namespaces();
+    auto it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto delegates = ns_bodies[0]->get_delegates();
+    REQUIRE(delegates.size() == 2);
+
+    REQUIRE(delegates.find("D1") != delegates.end());
+    REQUIRE(delegates.at("D1")->get_return_type() != std::nullopt);
+    auto del0_return_type = delegates.at("D1")->get_return_type()->get_semantic();
+    REQUIRE((del0_return_type.is_resolved() && std::get<simple_type>(del0_return_type.get_resolved_target()) == simple_type::Int32));
+    auto del0_formal_params = delegates.at("D1")->get_formal_parameters();
+    REQUIRE(del0_formal_params.size() == 3);
+    auto del0_formal_param_0_type = del0_formal_params[0].get_type().get_semantic();
+    auto del0_formal_param_1_type = del0_formal_params[1].get_type().get_semantic();
+    auto del0_formal_param_2_type = del0_formal_params[2].get_type().get_semantic();
+    REQUIRE(del0_formal_params[0].get_id() == "i");
+    REQUIRE(del0_formal_params[1].get_id() == "d");
+    REQUIRE(del0_formal_params[2].get_id() == "e");
+    REQUIRE((del0_formal_param_0_type.is_resolved() && std::get<simple_type>(del0_formal_param_0_type.get_resolved_target()) == simple_type::Int32));
+    REQUIRE((del0_formal_param_1_type.is_resolved() && std::get<simple_type>(del0_formal_param_1_type.get_resolved_target()) == simple_type::Double));
+    REQUIRE((!del0_formal_param_2_type.is_resolved() && del0_formal_param_2_type.get_ref_name() == "E"));
+
+    REQUIRE(delegates.find("D2") != delegates.end());
+    REQUIRE(delegates.at("D2")->get_return_type() == std::nullopt);
+    REQUIRE(delegates.at("D2")->get_formal_parameters().empty());
+}
+
+TEST_CASE("Property test")
+{
+    std::istringstream property_test_idl{ R"(
+        namespace N
+        {
+            interface I
+            {
+                Int32 property_1;
+                Int32 property_2 { get; }
+                Int32 property_3 { set; };
+                Int32 property_2 { set; };
+                Int32 property_3 { get; };
+                Int32 property_4 { get; set; };
+                Int32 property_5 { set; get; }
+            }
+
+            runtimeclass C
+            {
+                Int32 property_1;
+                Int32 property_2 { get; }
+                Int32 property_3 { set; };
+                Int32 property_2 { set; };
+                Int32 property_3 { get; };
+                Int32 property_4 { get; set; };
+                Int32 property_5 { set; get; }
+            }
+        }
+    )" };
+
+    std::istringstream incorrect_property_test_idl{ R"(
```

Comment:
This test is pretty big. Split off these "invalid" tests into their own test case, so they don't get lost in the sea of validation for valid properties.